### PR TITLE
Create branch

### PR DESCRIPTION
**Pull request by kumar88kanishk**
Create branch

### CODE REVIEW

File: .idea/ClojureProjectResolveSettings.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (divs1210) [inline .idea/ClojureProjectResolveSettings.xml]: remove this file

File: .idea/codeStyles/codeStyleConfig.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (divs1210) [inline .idea/codeStyles/codeStyleConfig.xml]: remove this file

File: .idea/hugit.iml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (divs1210) [inline .idea/hugit.iml]: remove this file

File: .idea/misc.xml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (divs1210) [inline .idea/misc.xml]: remove this file

File: .idea/vcs.xml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (divs1210) [inline .idea/vcs.xml]: remove this file

File: .idea/workspace.xml [hunk 6]
```diff
@@ -0,0 +1,360 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (divs1210) [inline .idea/workspace.xml]: remove this file

File: src/hugit/app/views.cljs [hunk 7]
```diff
@@ -304,28 +304,45 @@
...
        :window-size (- rows 6)
-       :items @branches
+       :items       @branches
        :item-props-f
-       (fn [branch]
-         (if (= @current-branch branch)
-           {:style {:fg :green}}
-           {:style {:fg :white}}))
-       :selected @selected
+                    (fn [branch]
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: fix indentation

File: src/hugit/app/views.cljs [hunk 8]
```diff
@@ -304,28 +304,45 @@
...
+       :selected    @selected
        :custom-key-handlers
-       {["enter"] {:f (fn [idx]
-                        (let [branch (nth @branches idx)]
-                          (toast> "Checking out " branch)
-                          (evt> [:checkout-branch branch])
-                          (evt> [:router/goto :status])))
-                   :label "Checkout"
-                   :type "Action"}}
+                    {["enter"] {:f     (fn [idx]
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: fix indentation

File: src/hugit/app/views.cljs [hunk 9]
```diff
@@ -304,28 +304,45 @@
...
+                                           (let [branch (nth @branches idx)]
+                                                (toast> "Checking out " branch)
+                                                (evt> [:checkout-branch branch])
+                                                (evt> [:router/goto :status])))
+                                :label "Checkout"
+                                :type  "Action"}
+                     ["n"]     {:f (fn [idx]
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: we don't need `idx`, make it `_`

File: src/hugit/app/views.cljs [hunk 10]
```diff
@@ -304,28 +304,45 @@
...
+                                       (evt> [:router/goto :input
+                                              {:label     "New Branch"
+                                               :on-submit (fn [branch-name]
+                                                              (print branch-name)
+                                                              (if (-> @(<sub [:repo :branches :local])
+                                                                      (set)
+                                                                      (contains? branch-name))
+                                                                ((toast> "Branch Already Exits")
+                                                                  (evt> [:router/goto :branches {:label "Local Branches"
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: no need to fire this event, they are already on this page

File: src/hugit/app/views.cljs [hunk 11]
```diff
@@ -304,28 +304,45 @@
...
+                                :type  "Action"}
+                     ["n"]     {:f (fn [idx]
+                                       (evt> [:router/goto :input
+                                              {:label     "New Branch"
+                                               :on-submit (fn [branch-name]
+                                                              (print branch-name)
+                                                              (if (-> @(<sub [:repo :branches :local])
+                                                                      (set)
+                                                                      (contains? branch-name))
+                                                                ((toast> "Branch Already Exits")
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: A branch called `<branch-name>` already exists!

Comment (divs1210) [inline src/hugit/app/views.cljs]: This doesn't seem right.To execute multiple statements one after another, use `do`:
```clojure
(do
  (expr1)
  (expr2))
```
and not
```clojure
((expr1)
 (expr2))
```

File: src/hugit/app/views.cljs [hunk 12]
```diff
@@ -378,3 +395,12 @@
       :width "100%"}
      [viewport (- rows 3)]
      [toast]]))
+
+(defn create-branch []
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: this function shouldn't exist

File: src/hugit/events.cljs [hunk 13]
```diff
@@ -237,3 +237,13 @@
      ;; TODO: implement
      )
    db))
+
+(rf/reg-event-db
+ :new-branch
```
Comment (divs1210) [inline src/hugit/events.cljs]: `:create-branch`

File: yarn.lock [hunk 14]
```diff
@@ -0,0 +1,3965 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (divs1210) [inline yarn.lock]: remove this file

File: .idea/ClojureProjectResolveSettings.xml (hunk 1 above)
Comment (kumar88kanishk) [reply .idea/ClojureProjectResolveSettings.xml]: @divs1210 this should be in gitignore, putting it in gitignore

File: src/hugit/app/views.cljs (hunk 10 above)
Comment (kumar88kanishk) [reply src/hugit/app/views.cljs]: If i don't call the view again the app hangs and no control works, neither any of them back/forward/enter/escape works.

**Comment by kumar88kanishk** [comment]
Changes added

File: src/hugit/app/views.cljs [hunk 15]
```diff
@@ -304,28 +304,45 @@
...
-       {["enter"] {:f (fn [idx]
-                        (let [branch (nth @branches idx)]
-                          (toast> "Checking out " branch)
-                          (evt> [:checkout-branch branch])
-                          (evt> [:router/goto :status])))
-                   :label "Checkout"
-                   :type "Action"}}
-       :on-back
-       #(evt> [:router/go-back])}]]))
+                    {["enter"] {:f (fn [idx]
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: fix indentation

File: src/hugit/events.cljs [hunk 16]
```diff
@@ -237,3 +237,13 @@
      ;; TODO: implement
      )
    db))
+
+(rf/reg-event-db
+ :create-branch
+ (fn [db [_ branch-name]]
+   (let [{:keys [command stdout stderr]}
```
Comment (divs1210) [inline src/hugit/events.cljs]: fix indentation

File: src/hugit/git.cljs [hunk 17]
```diff
@@ -189,3 +189,7 @@
        (-> line
            .trim
            (str/replace "* " ""))))))
+
+(defn create-branch [branch-name]
+  (println "Hugit.git new branch")
```
Comment (divs1210) [inline src/hugit/git.cljs]: remove the print

File: .idea/ClojureProjectResolveSettings.xml (hunk 1 above)
Comment (divs1210) [reply .idea/ClojureProjectResolveSettings.xml]: @kumar88kanishk also remove this file. it should not be committed.

File: src/hugit/app/views.cljs [hunk 18]
```diff
@@ -304,28 +304,45 @@
...
+                                             :on-submit (fn [branch-name]
+                                                            (print branch-name)
+                                                            (if (-> @(<sub [:repo :branches :local])
+                                                                    (set)
+                                                                    (contains? branch-name))
+                                                              (do
+                                                                (toast> (str "A branch called " branch-name " already exits!"))
+                                                                (evt> [:router/goto :branches {:label         "Local Branches"
+                                                                                               :branches-path [:repo :branches :local]}]))
+                                                              ((toast> "Creating new branch")
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: Fix the `((`

**Comment by kumar88kanishk** [comment]
Changes added

File: src/hugit/app/views.cljs (hunk 15 above)
Comment (divs1210) [reply src/hugit/app/views.cljs]: this indentation is incorrect. `:custom-key-handlers` and `{["enter"] {:f (fn [idx] ...` should be aligned.

File: src/hugit/events.cljs [hunk 19]
```diff
@@ -237,3 +237,14 @@
...
+          (git/create-branch branch-name)]
+      (do
```
Comment (divs1210) [inline src/hugit/events.cljs]: `let`s, `fn`s, and several other macros have implicit `do`s.

File: src/hugit/app/views.cljs [hunk 20]
```diff
@@ -304,28 +304,46 @@
...
-                        (let [branch (nth @branches idx)]
-                          (toast> "Checking out " branch)
-                          (evt> [:checkout-branch branch])
-                          (evt> [:router/goto :status])))
-                   :label "Checkout"
-                   :type "Action"}}
-       :on-back
-       #(evt> [:router/go-back])}]]))
+                    {["enter"] {:f (fn [idx]
+                                     (let [branch (nth @branches idx)]
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: the body of this `let` is incorrectly indented.

File: src/hugit/app/views.cljs [hunk 21]
```diff
@@ -304,28 +304,46 @@
...
+                                          (toast> "Checking out " branch)
+                                          (evt> [:checkout-branch branch])
+                                          (evt> [:router/goto :status])))
+                                :label "Checkout"
+                                :type  "Action"}
+                     ["n"]     {:f (fn [_]
+                                     (evt> [:router/goto :input
+                                            {:label     "New Branch"
+                                             :on-submit (fn [branch-name]
+                                                            (print branch-name)
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: the body of this `fn` is incorrectly indented.

Comment (divs1210) [inline src/hugit/app/views.cljs]: remove the `print`

File: src/hugit/app/views.cljs [hunk 22]
```diff
@@ -304,28 +304,46 @@
...
+                                                            (if (-> @(<sub [:repo :branches :local])
+                                                                    (set)
+                                                                    (contains? branch-name))
+                                                              (do
+                                                                (toast> (str "A branch called " branch-name " already exits!"))
+                                                                (evt> [:router/goto :branches {:label         "Local Branches"
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: put the view state on the next line.

File: src/hugit/app/views.cljs [hunk 23]
```diff
@@ -304,28 +304,46 @@
...
+                     ["n"]     {:f (fn [_]
+                                     (evt> [:router/goto :input
+                                            {:label     "New Branch"
+                                             :on-submit (fn [branch-name]
+                                                            (print branch-name)
+                                                            (if (-> @(<sub [:repo :branches :local])
+                                                                    (set)
+                                                                    (contains? branch-name))
+                                                              (do
+                                                                (toast> (str "A branch called " branch-name " already exits!"))
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: typo: `exists`, not `exits`

**Comment by kumar88kanishk** [comment]
Changes done. Indentation may break at your end. Intellij has no settings for cljs and neither default

File: src/hugit/app/views.cljs [hunk 24]
```diff
@@ -304,28 +304,46 @@
...
-       {["enter"] {:f (fn [idx]
-                        (let [branch (nth @branches idx)]
-                          (toast> "Checking out " branch)
-                          (evt> [:checkout-branch branch])
-                          (evt> [:router/goto :status])))
-                   :label "Checkout"
-                   :type "Action"}}
-       :on-back
-       #(evt> [:router/go-back])}]]))
+          {["enter"] {:f (fn [idx]
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: this is still one space off?

File: src/hugit/app/views.cljs (hunk 7 above)
Comment (divs1210) [reply src/hugit/app/views.cljs]: this hasn't been fixed?

File: src/hugit/app/views.cljs [hunk 25]
```diff
@@ -304,28 +304,46 @@
...
+                             (let [branch (nth @branches idx)]
+                                  (toast> "Checking out " branch)
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: fix indentation

File: src/hugit/app/views.cljs [hunk 26]
```diff
@@ -304,28 +304,46 @@
...
+                                  (evt> [:checkout-branch branch])
+                                  (evt> [:router/goto :status])))
+                      :label "Checkout"
+                      :type  "Action"}
+           ["n"] {:f (fn [_]
+                         (evt> [:router/goto :input
+                                {:label     "New Branch"
+                                 :on-submit (fn [branch-name]
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: bring `fn` to next line, never make files too wide.

**Comment by kumar88kanishk** [comment]
Fixed indentation

File: src/hugit/app/views.cljs [hunk 27]
```diff
@@ -304,28 +304,48 @@
...
+                                 (toast> "Checking out " branch)
+                                 (evt> [:checkout-branch branch])
+                                 (evt> [:router/goto :status])))
+                      :label "Checkout"
+                      :type  "Action"}
+           ["n"] {:f
+                    (fn [_]
+                        (evt> [:router/goto :input
+                               {:label     "New Branch"
+                                :on-submit (fn [branch-name]
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: put `fn` on next line

File: src/hugit/app/views.cljs [hunk 28]
```diff
@@ -304,28 +304,48 @@
...
-       {["enter"] {:f (fn [idx]
-                        (let [branch (nth @branches idx)]
-                          (toast> "Checking out " branch)
-                          (evt> [:checkout-branch branch])
-                          (evt> [:router/goto :status])))
-                   :label "Checkout"
-                   :type "Action"}}
-       :on-back
-       #(evt> [:router/go-back])}]]))
+          {["enter"] {:f
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: fix indentation

File: src/hugit/app/views.cljs [hunk 29]
```diff
@@ -304,28 +304,48 @@
...
+                        (fn [idx]
+                            (let [branch (nth @branches idx)]
```
Comment (divs1210) [inline src/hugit/app/views.cljs]: fix indentation

**Pull request merged**